tests/app/main/views/test_agreements.py

```diff
@@ -347,3 +354,17 @@ def test_status_filtered_final_with_status_redirects_to_list(self, data_api_clie
     def test_invalid_status_raises_400(self, data_api_client):
         response = self.client.get('/admin/suppliers/151/agreements/g-cloud-8/next?status=bad')
         assert response.status_code == 400
+
+    @pytest.mark.parametrize("role,expected_code", [
+        ("admin", 403),
+        ("admin-ccs-category", 403),
+        ("admin-ccs-sourcing", 302),
+        ("admin-framework-manager", 302),
+        ("admin-manager", 403),
+    ])
+    def test_next_agreement_redirect_only_accessible_to_specific_user_roles(self, data_api_client, role, expected_code):
```

Comment:
Hmm, thinking about this the "next agreement" should maybe only work for sourcing, as it is the redirect target after a successful "POST" to the agreements page (to approve or put on hold), so "view only" roles should never hit the "next agreement" route as far as I can think.

Reply:
The 'next agreement' URL is on the `view_signed_agreement` template (i.e. as a link, not a form action)?

Reply:
Ah is it? OK ignore me then.